xbmc/filesystem/Directory.cpp

```diff
@@ -305,6 +305,45 @@ bool CDirectory::GetDirectory(const CURL& url,
   return false;
 }
 
+bool CDirectory::EnumerateDirectory(const std::string& strPath,
+                                    DirectoryEnumerationCallback callback,
+                                    bool bFileOnly /* = false */,
+                                    const std::string &strMask /* = "" */,
+                                    int flags /* = DIR_FLAG_DEFAULTS */)
+{
+  CFileItemList items;
+
+  // get items in specified directory
+  if (!CDirectory::GetDirectory(strPath, items, strMask, flags))
+    return false;
+
+  // process all files
+  for (int i = 0; i < items.Size(); ++i)
+  {
+    auto item = std::make_shared<CFileItem>(*items[i]);
```

Comment:
This could be `const auto ...`. I know this is not part of the coding guidelines, but common practice, to make intention clear that this var will not be changed anymore.

Maybe you want to change this here and everywhere else in the PR.

And regarding modern c++, we should always initialize instances using curly braces, not assignment or normal braces. There are long write-ups why curly braces are the only safe way to initialize.

=> `const auto item{std::make_shared<CFileItem>(*items[i])};`